### PR TITLE
[MMB-209] New Layout for Avatar Stack

### PR DESCRIPTION
**Pull request by artismarti**
Designs here: https://www.figma.com/file/sG6lgro7ruOkBi9D2XoyQM/Examples-Pages?type=design&node-id=1-3&mode=design&t=b44HxOIzhk2eAJMN-0

### CODE REVIEW

File: src/components/AvatarStack/AvatarStack.tsx [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 
 /** 💡 Select a mock name to assign randomly to a new user that enters the space💡 */
 const mockName = () => mockNames[Math.floor(Math.random() * mockNames.length)];
+const avatarColor = () =>
+  avatarColors[Math.floor(Math.random() * avatarColors.length)];
 
 const AvatarStack = ({ spaceName }: { spaceName: string }) => {
   const [members, setMembers] = useState<SpaceMember[]>([]);
   const name = useMemo(mockName, []);
+  const memberColor = useMemo(avatarColor, []);
```
Comment (snikidev) [inline src/components/AvatarStack/AvatarStack.tsx:20]: You sure we need `useMemo` here? I believe you! 😄

File: src/components/AvatarStack/Avatars.tsx [hunk 2]
```diff
@@ -1,66 +1,95 @@
...
-            otherUsers.length > MAX_USERS_BEFORE_LIST
-              ? (index + 1) * HORIZONTAL_SPACING_OFFSET
-              : index * HORIZONTAL_SPACING_OFFSET;
-          return (
+      {users.map((user, index) => {
+        const HORIZONTAL_SPACING_OFFSET = 40;
+        const rightOffset =
+          users.length > MAX_USERS_BEFORE_LIST
+            ? (index + 1) * HORIZONTAL_SPACING_OFFSET
+            : index * HORIZONTAL_SPACING_OFFSET;
```
Comment (snikidev) [inline src/components/AvatarStack/Avatars.tsx:27]: I like to put staff like that in a util function. But that's just me, not necessary really 🙃

File: src/components/AvatarStack/Avatars.tsx [hunk 3]
```diff
@@ -1,66 +1,95 @@
...
+              } h-12 w-12 rounded-full flex items-center justify-center relative border-2 border-gray-200`}
+              onMouseOver={() => setHoveredClientId(user.clientId)}
+              onMouseLeave={() => setHoveredClientId(null)}
+              id="avatar"
             >
-              <UserCircleIcon className="absolute mt-2 h-8 w-8 opacity-80 stroke-white fill-transparent pointer-events-none" />
+              <p
+                className={`relative z-20 text-xs ${
+                  user.isConnected ? "text-white" : "text-gray-400"
+                }`}
```
Comment (snikidev) [inline src/components/AvatarStack/Avatars.tsx:49]: don't we have clsx in this repo? (https://www.npmjs.com/package/clsx) Big fan! I find classnames are easier to read that way
```
classNames={clsx('relative z-20 text-xs', { "text-white": user.isConnected, "text-gray-400": !user.isConnected })}
```
But again, no biggie 👍

File: src/components/AvatarStack/Avatars.tsx [hunk 4]
```diff
@@ -1,66 +1,95 @@
...
 };
+const Avatars = ({ otherUsers }: { otherUsers: SpaceMember[] }) => {
+  const OVERLAP_AMOUNT = 40;
+  const AVATAR_WIDTH = 48;
+  const totalWidth = AVATAR_WIDTH + OVERLAP_AMOUNT * otherUsers.length;
+
+  return (
+    <div
+      className="relative flex max-w-[206px] "
+      style={{ width: `${totalWidth}px` }}
```
Comment (snikidev) [inline src/components/AvatarStack/Avatars.tsx:83]: do we actually need `px` here, I think we can just do the number and it will do the px on itself

**Review by snikidev — APPROVED** [review]
Looks good! Everything I said is just me talking about me and my personal preferences 😁

File: src/components/AvatarStack/Avatars.tsx (hunk 3 above)
Comment (artismarti) [reply src/components/AvatarStack/Avatars.tsx:49]: will skip this one :)

File: src/components/AvatarStack/Avatars.tsx (hunk 4 above)
Comment (artismarti) [reply src/components/AvatarStack/Avatars.tsx:83]: afaik, this is needed unless you're doing it in plain HTML/CSS. With React & Vanilla JS you do need to explicitly state it.

**Pull request merged**